### PR TITLE
Create decorator to read in multiple files with pattern

### PR DESCRIPTION
**Pull request by samarpan-rai**
Re-did the old PR #47 because it was messy and most importantly after my local merge `black` kept failing in the CI/test so opened a new PR with a new fork.

### CODE REVIEW

**Comment by samukweku** [comment]
Quiet On this function. Looks good and expanded my knowledge of decorators

File: tests/test_decorator/test_multifile.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    Test that csv files can be read given a pattern
+    """
+
+    writer = Clumper.read_csv("tests/data/monopoly.csv")
+
+    for i in range(copies):
+        writer.write_csv(tmp_path / f"monopoly_copy_{i}.csv")
+
+    reader = Clumper.read_csv(str(tmp_path / "*.csv"))
+    assert len(reader) == (copies) * len(writer)
```
Comment (koaning) [inline tests/test_decorator/test_multifile.py:52]: These three tests are doing *nearly* the same thing. Might we be able to parametrize them? 

If you have something like; 

```
[(Clumper.read_csv, "tests/data/monopoly.csv"), (Clumper.read_jsonl, "tests/data/cards.jsonl"), ...])
```

Then you might be able to generalize this.

File: tests/test_decorator/test_multifile.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    Test that jsonl files can be read given a pattern
+    """
+
+    writer = Clumper.read_jsonl("tests/data/cards.jsonl")
+
+    for i in range(copies):
+        writer.write_jsonl(tmp_path / f"cards_copy_{i}.jsonl")
+
+    reader = Clumper.read_jsonl(str(tmp_path / "*.jsonl"))
+    assert len(reader) == (copies) * len(writer)
```
Comment (koaning) [inline tests/test_decorator/test_multifile.py:22]: `(copies)` -> `copies`

**Review by koaning — CHANGES_REQUESTED** [review]
I have a minor request on the tests. There's some redundant characters. Other than that ... it looks good! I'll need to find some time to play with this locally but from looking at it -> looks great! 

@samarpan-rai @samukweku thanks!

**Comment by koaning** [comment]
I'm reminded of something else. We need to add this feature to the docstrings!

**Comment by samarpan-rai** [comment]
@koaning thank you for all the feedbacks. Should the docstring be a separate PR? I think adding documentation after merge would be neater but it's upto you. I can easily add documentation in the same PR.

**Comment by koaning** [comment]
Docstring in a separate PR is fine.

**Comment by koaning** [comment]
@samarpan-rai just to check, I can merge this?

**Comment by samarpan-rai** [comment]
> @samarpan-rai just to check, I can merge this?

Yes.

> @samarpan-rai just to check, I can merge this?

Yes.

**Pull request merged**